### PR TITLE
Update solution to lecture1

### PR DESCRIPTION
**Pull request by TseEnSun**
### Update the solutions for Lecture 1

Hi, @chshersh. Thanks for your review. I updated the code and took your suggestion. Please help to review it again. Thank you.

### CODE REVIEW

File: src/Lecture1.hs [hunk 1]
```diff
@@ -121,21 +119,8 @@ and finds a sum of the numbers inside this string.
...
-    | otherwise = split c (acc ++ [x]) xs
-
-sumStr :: [String] -> Int
-sumStr [] = 0
-sumStr (x:xs) = (read x :: Int) + sumStr xs
-
 strSum :: String -> Int
-strSum "" = 0
-strSum str = sumStr $ split ' ' "" str
+strSum str = sum $ map (\x -> read x :: Int) $ words str
```
Comment (chshersh) [inline src/Lecture1.hs:124]: As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
strSum str = sum (map read (words str))
```

File: src/Lecture1.hs [hunk 2]
```diff
@@ -160,5 +145,10 @@ lowerAndGreater n list =
...
         where
-            gls = length $ filter (< n) list
-            lts = length $ filter (> n) list
+            (gls, lts) = foldl (
+                \(a, b) -> 
+                    \x -> 
+                        if x < n then (a + 1, b) 
+                        else if x > n then (a, b + 1)
+                        else (a, b)
+            ) (0, 0) list
```
Comment (chshersh) [inline src/Lecture1.hs]: GHC still produces a compilation error. Not sure why but try indenting the `)` a little bit further
```suggestion
                ) (0, 0) list
```

**Review by chshersh — APPROVED** [review]
The updates look great! 🎉

File: src/Lecture1.hs (hunk 1 above)
Comment (TseEnSun) [reply src/Lecture1.hs:124]: Wow, that looks fantastic. I was struggling with how to combine `read x :: Int` with the map function. 😆

**Pull request merged**